pkg/controller/association/secret.go

```diff
@@ -189,3 +193,21 @@ func filterManagedElasticRef(associations []commonv1.Association) []commonv1.Ass
 	}
 	return r
 }
+
+// copySecret will copy the source secret to the target namespace adding labels from the associated object to ensure garbage collection happens.
+func copySecret(ctx context.Context, client k8s.Client, secHash hash.Hash, targetNamespace string, source types.NamespacedName) error {
+	var original corev1.Secret
+	if err := client.Get(ctx, source, &original); err != nil {
+		return err
+	}
+	var expected corev1.Secret
+	expected.Data = original.Data
+	commonhash.WriteHashObject(secHash, original.Data)
+	expected.Labels = original.Labels
+	expected.Annotations = original.Annotations
+	expected.Name = original.Name
+	expected.Namespace = targetNamespace
+
+	_, err := reconciler.ReconcileSecret(ctx, client, expected, nil)
```

Comment:
I was wondering if we could use a soft owner to garbage collect this Secret.

```suggestion
	_, err := reconciler.ReconcileSecretNoOwnerRef(ctx, client, expected, softOwner)
```

`softOwner` being `referencedObj` IIUC, and maybe use `GarbageCollectSoftOwnedSecrets` to remove them (not tested, only an idea).

Reply:
I think garbage collection is already tied to the association, in this case the Fleet Server->ES association. This has the odd side effect that the secret stays if the Agent->FleetServer association is severed but the Fleet Server->ES association still exists. I was thinking that this would be sufficient though because typically you would deploy Fleet server and Agent in tandem.

Reply:
> This has the odd side effect that the secret stays if the Agent->FleetServer association is severed but the Fleet Server->ES association still exists.

Yes, this is what I was trying to "fix" with this suggestion. Not a strong opinion though, we can stick with the current behavior 👍